modules/callcenter/src/lib.rs

```diff
@@ -33,6 +36,10 @@ impl Callcenter {
     pub fn increment_counter(self: &mut State<Self>, counter_id: ModuleId) {
         self.transact(counter_id, "increment", ())
     }
+
+    pub fn calling_self(self: &mut State<Self>, id: ModuleId) -> bool {
+        self_id() == &id
```

Comment:
also, i think in case of calling top-level functions, a `dallo::self_id()` will communicate more clearly what's going on.